net: wifi: shell:  Correct help text for `connect` command

rado17 opened the pull request:
Correct the information provided in help text for key management.

rado17 commented:
@sachinthegreen @ajayparida @VivekUppunda @kapbh , Please have a look.

krish2718 commented on subsys/net/l2/wifi/wifi_shell.c:
```diff
@@ -501,6 +501,7 @@ static int __wifi_args_to_params(const struct shell *sh, size_t argc, char *argv
 				params->psk_length = strlen(params->psk);
 			} else {
 				PR_WARNING("Passphrase provided without security configuration\n");
+				return -EINVAL;
```
This puts a dependency on the order, e.g., if `-p X -k Y` it would fail with `Passphrase provided without security configuration`

krish2718 reviewed (commented):
LGTM. Please update the commit log.

rlubos commented:
I recall we have an alternate similar PR already open:
https://github.com/zephyrproject-rtos/zephyr/pull/71007/files

In there, the warning was moved after all options has been processed, to avoid ordering issues.

rado17 commented:
> I recall we have an alternate similar PR already open: https://github.com/zephyrproject-rtos/zephyr/pull/71007/files
> 
> In there, the warning was moved after all options has been processed, to avoid ordering issues.

I will drop the change related to the warning from this PR since PR#71007 is already taking care of that.

rado17 committed:
net: wifi: shell: Correct help text for `connect` command

Since `NONE` is a valid option for key management in `connect`
command, we should remove the part in help text which says:
"valid only for secure SSIDs".

Signed-off-by: Ravi Dondaputi <ravi.dondaputi@nordicsemi.no>

krish2718 reviewed (dismissed):
that comment is still valid, just that we are doing an extra check now, so, might as well leave it?

rado17 committed:
net: wifi: shell: Remove redundant break

Having a `break` statement after `return` is redundant and
needs to be removed.

Signed-off-by: Ravi Dondaputi <ravi.dondaputi@nordicsemi.no>
